mountpoint-s3/src/fs/handles.rs

```diff
@@ -329,16 +327,23 @@ where
         result
     }
 
-    pub async fn complete_if_in_progress(self, key: &str) -> Result<(), Error> {
+    /// Check state of upload, and complete the upload if it is still in-progress.
+    ///
+    /// When successful, returns `true` where the upload was still in-progress and thus completed by this method call.
+    pub async fn complete_if_in_progress(self, key: &str) -> Result<bool, Error> {
         match self {
             UploadState::AppendInProgress {
                 request,
                 handle,
                 initial_etag,
                 ..
-            } => Self::complete_append(request, key, handle, initial_etag).await,
-            UploadState::MPUInProgress { request, handle, .. } => Self::complete_upload(request, key, handle).await,
-            UploadState::Failed(_) | UploadState::Completed => Ok(()),
+            } => Self::complete_append(request, key, handle, initial_etag)
+                .await
+                .map(|()| true),
+            UploadState::MPUInProgress { request, handle, .. } => {
+                Self::complete_upload(request, key, handle).await.map(|()| true)
+            }
```

Comment:
nit: I'd prefer something like this:
```suggestion
            } => {
              Self::complete_append(request, key, handle, initial_etag).await?;
                Ok(true)
              }
            UploadState::MPUInProgress { request, handle, .. } => {
                Self::complete_upload(request, key, handle).await?;
                Ok(true)
            }
```

Reply:
That's a lot nicer, thanks